### PR TITLE
COVIDNet-P

### PR DESCRIPTION
**Pull request by NaomiTerhljan**
# Pull Request Template

## Description

Created COVIDNet-P inference and eval scripts. Added a README for COVIDNet-P which links from the main README page. 

## Context of change

Please add options that are relevant and mark any boxes that apply.

- [ ] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [x] Tool (tool that assists coding development)
- [ ] Other

## Type of change

Please mark any boxes that apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Checklist:

Please mark any boxes that have been completed.

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ If you find our work useful, can cite our paper using:
 4. COVIDx-CXR dataset: https://github.com/lindawangg/COVID-Net/blob/master/docs/COVIDx.md
 5. COVIDx-CT dataset: https://github.com/haydengunraj/COVIDNet-CT/blob/master/docs/dataset.md
 6. COVIDx-S dataset: https://github.com/lindawangg/COVID-Net/tree/master/annotations
+7. COVIDNet-P inference for pneumonia: https://github.com/lindawangg/COVID-Net/blob/master/docs/covidnet_pneumonia.md
```
Comment (mayaliliya) [inline README.md:63]: we should also add an update to the readme above such as "We released COVIDNet-P [inference] (inference_pneumonia.py) and [evaluation] (eval_pneumonia.py) scripts for identifying pneumonia in CXR images using COVIDx4 dataset. For more information please refer to this [doc] (docs/covidnet_pneumonia.md)."

File: docs/covidnet_pneumonia.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Inference for Pneumonia Cases
+
+This section describes how the currently available models are able to be leveraged for use distinguishing only between cases where
```
Comment (mayaliliya) [inline docs/covidnet_pneumonia.md:3]: Might want to add in which available models as well provide links to the model docs by using the square bracket style, i.e. 
The COVIDNet Chest CXR [models] (docs/covidnet_pneumonia.md) can be leveraged to distinguish ....

File: docs/covidnet_pneumonia.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# Inference for Pneumonia Cases
```
Comment (mayaliliya) [inline docs/covidnet_pneumonia.md:1]: Evaluation and Inference for Pneumonia Cases

File: docs/covidnet_pneumonia.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+This section describes how the currently available models are able to be leveraged for use distinguishing only between cases where
+pneumonia is present and where it is not. The file inference_pneumonia.py makes use of COVIDNet-CXR models and modifies the output to return a prediction of whether
+pneumonia is present or not in a given image.
```
Comment (mayaliliya) [inline docs/covidnet_pneumonia.md:5]: Maybe add in a small blurb on how the models works.
The COVIDNet-P evaluation scripts take as input an image of shape (N, 480, 480, 3) and output the softmax probability of pneumonia by leveraging the predicted output from the COVIDNet-CXR4 models.

File: docs/covidnet_pneumonia.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
```
Comment (mayaliliya) [inline docs/covidnet_pneumonia.md:6]: We should add a section for "Steps for Evaluation" as well since we are adding the scipt

File: eval_pneumonia.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import tensorflow as tf
+import os, argparse
+import cv2
+
+from data import process_image_file
+
+mapping = {'normal': 0, 'pneumonia': 1, 'COVID-19': 1}
```
Comment (mayaliliya) [inline eval_pneumonia.py:9]: maybe just add a comment like "# Combine the COVID cases and non-COVID pneumonia cases" just for people to understand our mapping idea

File: inference_pneumonia.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+parser.add_argument('--ckptname', default='model-18540', type=str, help='Name of model ckpts')
+parser.add_argument('--imagepath', default='assets/ex-covid.jpeg', type=str, help='Full path to image to be inferenced')
+parser.add_argument('--in_tensorname', default='input_1:0', type=str, help='Name of input tensor to graph')
+parser.add_argument('--out_tensorname', default='norm_dense_1/Softmax:0', type=str, help='Name of output tensor from graph')
+parser.add_argument('--input_size', default=480, type=int, help='Size of input (ex: if 480x480, --input_size 480)')
+parser.add_argument('--top_percent', default=0.08, type=float, help='Percent top crop from top of image')
+
+args = parser.parse_args()
+
+mapping = {'normal': 0, 'pneumonia': 1, 'pneumonia': 2}
```
Comment (mayaliliya) [inline inference_pneumonia.py:20]: similar here, maybe add a comment like "# Combine the COVID and non-COVID pneumonia predictions"

File: README.md [hunk 8]
```diff
@@ -4,7 +4,8 @@
 
 **Recording to webinar on [How we built COVID-Net in 7 days with Gensynth](https://darwinai.news/fny)**
 
-**Update 11/09/2020** We released updated COVIDNet-S [models](docs/models.md) for geographic and opacity extent scoring of SARS-CoV-2 lung severity and updated the [inference script](docs/covidnet_severity.md) for an opacity extent scoring ranging from 0-8.
+**Update 11/15/2020** We released COVIDNet-P [inference](inference_pneumonia.py) and [evaluation](eval_pneumonia.py) scripts for identifying pneumonia in CXR images using the COVIDx4 dataset. For more information please refer to this [doc](docs/covidnet_pneumonia.md).
+**Update 09/11/2020** We released updated COVIDNet-S [models](docs/models.md) for geographic and opacity extent scoring of SARS-CoV-2 lung severity and updated the [inference script](docs/covidnet_severity.md) for an opacity extent scoring ranging from 0-8.
```
Comment (NaomiTerhljan) [inline README.md:8]: The date for the removed line was formatted wrong.. all other entries are month/day/year, this one was day/month/year so I fixed it.

**Review by mayaliliya — APPROVED** [review]
